### DISCUSSION TITLE
2D map of simulated data

### DISCUSSION THREAD

**Discussion by yuehengzhang95** [mtex-toolbox/mtex#1356] [Ask Anything] [answered]
Hi Rüdiger,

I am curious if we can simulate the 2D orientation map with inverse pole figure color for my simulated dataset? The purpose is for me to compare the EBSD map with my simulated data. The tricky thing is that how we can use the euler angles to get the corresponding RGB values for coloring? I know for EBSD, we just need to use the following lines of code:
`%%
% % cs = crystalSymmetry('432', [4.1 4.1 4.1], 'mineral', 'Gold', 'color', [0.53 0.81 0.98])
oM = ipfHSVKey(ebsd('Gold'))
% % 
oM.inversePoleFigureDirection = yvector; # if we want to plot the map based on the sample x direction(010) 
plot(oM)
color = oM.orientation2color(ebsd('Gold').orientations);`

However, I am not sure how to get color for my simulated data.

The format of the simulated data is [[x1,y1,euler1a,euler1b,euler1c],[x2,y2,euler2a,euler2b,euler2c],[x3,y3,euler3a,euler3b,euler3c],....].x_i and y_i are coordinates of the orientations(represented as Euler angles, which is the third, fourth and fifth column - euler1a, euler1b, euler1c etc..).

PS. if we want to plot the map based on the sample x direction(010) instead of Z direction(001), what else do I need to do. Many thanks!

Yueheng

**Comment by kilir** [comment, marked as answer]
Hi,
have a look at the help for `loadEBSD_generic`. For the seelction of the reference direction see the property .inversePoleFigureDirection` https://mtex-toolbox.github.io/EBSDIPFMap.html
Cheers,
Rüdiger
